src/tox/tox_env/python/pip/pip_install.py

```diff
@@ -100,9 +133,20 @@ def _install_requirement_file(self, arguments: PythonDeps, section: str, of_type
                     missing_requirement = set(old["requirements"]) - set(new_requirements)
                     if missing_requirement:
                         raise Recreate(f"requirements removed: {' '.join(missing_requirement)}")
+                    if old.get("constraint_options") != constraint_options:
+                        raise Recreate(
+                            "constraint options changed: old={} new={}".format(
```

Comment:
why is this not an f-string?

Reply:
mainly because i'm calling `old.get("constraint_options"),`, but let me do that on the line above.

Reply:
can still do that in f-string just use ' instead?